### PR TITLE
Add support for WooCommerce Payments ad banner on extensions screen's "Featured" section

### PR DESCRIPTION
**Pull request by andfinally**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:
We want to be able to show a banner on the wc-admin featured extensions page, to highlight the benefits WooCommerce Pay offers to store owners and customers.

- Add support for a banner on the featured extensions section.
- Add CSS rules for the banner to `admin.scss`.

### WooCommerce.com issue

This change is related to 9860-gh-Automattic/woocommerce.com.

### How to test the changes in this Pull Request:

1. Check the branch out locally and build assets.
2. If the update to the WooCommerce.com featured endpoint  in 10179-gh-Automattic/woocommerce.com hasn't been merged yet, you'll need to modify the featured endpoint locally to see the ad. Copy the JSON from [here](293e7-pb/) and make it available at some publicly-accessible URL, like a GitHub gist. Go to `includes/admin/class-wc-admin-addons.php` and replace

```php
$raw_featured = wp_safe_remote_get(
	'https://woocommerce.com/wp-json/wccom-extensions/1.0/featured',
	array(
		'headers'    => $headers,
		'user-agent' => 'WooCommerce Addons Page',
	)
);
```

with

```php
$raw_featured = wp_safe_remote_get(
	[ URL for the file ],
);
```

You may need to flush the object cache and featured transient to see the changes:

```
wp cache flush
wp transient delete wc_addons_featured
```
3. To see the ad, WCPay should not be active on your test site, and your location should be in the US, Canada, UK, Ireland, Australia or New Zealand. If you're not in one of these countries, you can cheat by commenting out lines 793-795 of `includes/admin/class-wc-admin-addons.php`:

```php
if ( isset( $item->geowhitelist ) && ! in_array( $location, $item->geowhitelist, true ) ) {
    return false;
}
```

4. Go to the WooCommerce > Extensions page in your test site. You should see the "featured" section (`/wp-admin/admin.php?page=wc-addons&section=_featured`). Check that the ad appears as expected, and the links work.
5. Install and activate WCPay. The ad should not appear.
6. For bonus points, view the screen from an unsupported location, or hack the `show_extension` method so it doesn't like your country. The ad should not appear.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

New - Feature WooCommerce Payments in the extensions store for stores in the US, Canada, UK, Ireland, Australia and New Zealand.

### CODE REVIEW

**Review by vedanshujain — DISMISSED** [review]
Let some minor comments, otherwise testing well and looks good! Great work!

**Comment by andfinally** [comment]
Many thanks for taking a look @vedanshujain, I've hopefully addressed your points. I found the `!important` wasn't necessary after all. I saved an updated version of the test json in 293ca-pb/.

File: includes/admin/class-wc-admin-addons.php [hunk 1]
```diff
@@ -534,6 +534,69 @@ class="addons-img"
...
+		if (
+			! current_user_can( 'install_plugins' ) ||
+			! current_user_can( 'activate_plugins' )
+		) {
+			return;
+		}
+
+		$section_object = (object) $section;
+
+		if ( ! empty( $section_object->geowhitelist ) ) {
```
[the remark this answers is not in the document]
Comment (andfinally) [reply includes/admin/class-wc-admin-addons.php]: Thanks, changed in de331ad8935d01eafc910d84fc4464469cfd1dac.

File: assets/css/admin.scss [hunk 2]
```diff
@@ -364,6 +401,12 @@
 		color: #fff;
 	}
 
+	.addons-button-expandable {
+		display: inline-block;
+		padding: 0 16px;
+		width: auto !important;
```
[the remark this answers is not in the document]
Comment (andfinally) [reply assets/css/admin.scss]: I've removed this in de331ad8935d01eafc910d84fc4464469cfd1dac – the `!important` wasn't needed after all.

File: includes/admin/class-wc-admin-addons.php [hunk 3]
```diff
@@ -722,11 +788,11 @@ public static function install_woocommerce_payments_addon( $section = '_featured
 	 */
 	public static function show_extension( $item ) {
 		$location = WC()->countries->get_base_country();
-		if ( isset( $item->geowhitelist ) && ! in_array( $location, $item->geowhitelist, true ) ) {
+		if ( isset( $item->geo_allow_list ) && ! in_array( $location, $item->geo_allow_list, true ) ) {
```
Comment (vedanshujain) [inline includes/admin/class-wc-admin-addons.php:791]: hi, sorry, i did not realize that these naming are also being used here as well, which is being used for other items as well and not just wc pay. we should revert this change, because we don't know if all the other promotional extensions have changed the wording as well, and they have not then these wordings will not work for them anymore.

File: includes/admin/class-wc-admin-addons.php [hunk 4]
```diff
@@ -534,6 +534,69 @@ class="addons-img"
...
+		if (
+			! current_user_can( 'install_plugins' ) ||
+			! current_user_can( 'activate_plugins' )
+		) {
+			return;
+		}
+
+		$section_object = (object) $section;
+
+		if ( ! empty( $section_object->geo_allow_list ) ) {
```
Comment (vedanshujain) [inline includes/admin/class-wc-admin-addons.php:552]: should not we do the same for block list as well? as of now blocklist won't won't work

**Review by vedanshujain — COMMENTED** [review]
Looks like we may want to revert one of the feedback we addressed because of something I missed earlier. Sorry about that!

File: includes/admin/class-wc-admin-addons.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (andfinally) [reply includes/admin/class-wc-admin-addons.php:552]: Reverted in 6bb1ec73a4757cf38ddb1893a06515ccb6263b57.

File: includes/admin/class-wc-admin-addons.php (hunk 4 above)
Comment (andfinally) [reply includes/admin/class-wc-admin-addons.php]: Good point – added in 7202f19631e474d1adc93d83acc1499bab0b6b3d.

**Comment by andfinally** [comment]
Thanks Vedanshu, I've reverted the rename. I did search for those names in the WCCOM codebase and didn't find them in any other endpoints, so it would probably have been OK. But it's good to be cautious.

**Review by vedanshujain — APPROVED** [review]
LGTM!
